File: fix_109_header.js
const fs = require('fs');
let c = fs.readFileSync('herramienta-nutricion/src/components/Header.jsx', 'utf8');

// Fix 1: The export default declaration
const fix1 = `<<<<<<< HEAD
=======
export default function Header({ currentView, setCurrentView, selectedWeek, setSelectedWeek, currentUser, isAdmin, onOpenLogin, onLogout }) {
>>>>>>> origin/main`;

c = c.replace(fix1, "");
c = c.replace("currentUser,", "currentUser,\n  isAdmin,");

// Fix 2: The UI switchers
// They look like:
// <<<<<<< HEAD
//         {/* Service Model Switcher (Corporativo B2B vs Residencia de Mayores) */}
//         {onServiceProfileChange && (
// =======
//         {/* Selector de Herramientas EXCLUSIVO para Administradores */}
//         {isAdmin && (
// >>>>>>> origin/main
//           <div style={{
//             display: 'flex',
//             alignItems: 'center',
//             background: '#F1F5F9',
// <<<<<<< HEAD
//             padding: '0.2rem',
//             borderRadius: '9999px',
// ...
// =======
// ...
// >>>>>>> origin/main
//              </button>
//            </div>
//          )}

// Instead of trying to parse all that, since we just want BOTH, we can find the HEAD block and the MAIN block and just concatenate them!
const raw = fs.readFileSync('herramienta-nutricion/src/components/Header.jsx', 'utf8');
let newC = "";
let inConflict = false;
let currentBlock = [];

// A simpler way: we know what HEAD has, and we know what MAIN has.

File: fix_109_header.py
import re

with open('herramienta-nutricion/src/components/Header.jsx', 'r') as f:
    c = f.read()

# Fix the function signature
c = re.sub(r'<<<<<<< HEAD\n=======\nexport default function Header[^\n]+\n>>>>>>> origin/main\n', '', c)
c = c.replace('  currentUser,\n', '  currentUser,\n  isAdmin,\n')

# Fix the getRoleBadge
c = re.sub(r'<<<<<<< HEAD\n=======\n    if \(isAdmin\).*?\n>>>>>>> origin/main\n', "    if (isAdmin) return { label: 'Administrador', color: '#7C3AED', bg: '#F5F3FF' };\n", c)

# For the switchers, we want both. The structure is complex.
# We will just remove the conflict markers and let both render.
# Wait, the second conflict block splits the `div` style. 
# It's better to just extract HEAD's switcher and MAIN's switcher.

# Let's just remove the markers and fix the syntax manually if needed.
c = c.replace('<<<<<<< HEAD\n', '')
c = c.replace('=======\n', '')
c = c.replace('>>>>>>> origin/main\n', '')

with open('herramienta-nutricion/src/components/Header.jsx', 'w') as f:
    f.write(c)

